packages/astro/src/runtime/server/hydration.ts

```diff
@@ -140,12 +140,12 @@ export async function generateHydrateScript(
 	}
 
 	// Add component url
-	island.props['component-url'] = await result.resolve(componentUrl);
+	island.props['component-url'] = await result.resolve(decodeURI(componentUrl));
```

Comment:
`componentUrl` is retrieved as-is from the attribute, which is encoded. In code, we want to deal with the decoded version instead.